apps/financial-aid/web-osk/src/components/AppLayout/AppLayout.tsx

```diff
@@ -11,18 +12,48 @@ import {
 
 import cn from 'classnames'
 import { AppContext } from '@island.is/financial-aid-web/osk/src/components/AppProvider/AppProvider'
+import { Routes } from '@island.is/financial-aid/shared/lib'
 
 interface Props {
   children: ReactNode
 }
 
 const AppLayout = ({ children }: Props) => {
   const { isAuthenticated, user, loadingUser } = useContext(AppContext)
+  const router = useRouter()
 
   useEffect(() => {
     document.title = 'Fjárhagsaðstoð'
   }, [])
 
+  const getRedirectPath = () => {
+    if (user?.spouse?.hasFiles) {
+      return `${Routes.statusPage(user.currentApplicationId as string)}`
+    }
+    if (user?.spouse?.hasPartnerApplied) {
+      return `${Routes.form.info}`
+    }
+
+    if (user?.currentApplicationId) {
+      return `${Routes.statusPage(user.currentApplicationId as string)}`
+    }
+
+    return `${Routes.application}`
+  }
+
+  if (isAuthenticated && user) {
+    const redirectPath = getRedirectPath()
+
+    if (
+      router.pathname === '/' ||
+      (router.pathname.startsWith(Routes.application) &&
```

Comment:
I would probably start with having this in the if check on line 44 and would also draw that out as a variable for readability. Since we are checking for 4 things then it can be nice to draw it out.